renku_notebooks/api/classes/cloud_storage/existing.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+from ....config import config
+
+
+@dataclass
+class ExistingCloudStorage:
+    bucket: str
+    endpoint: str
+
+    @classmethod
+    def from_manifest(cls, manifest: Dict[str, Any]) -> List["ExistingCloudStorage"]:
+        output: List[ExistingCloudStorage] = []
+        endpoint_annotation = (
+            config.session_get_endpoint_annotations.renku_annotation_prefix + "endpoint"
+        )
+        for patch_collection in manifest["spec"]["patches"]:
+            for patch in patch_collection["patch"]:
+                if patch["op"] == "test":
+                    continue
+                if (
+                    isinstance(patch["value"], dict)
+                    and patch["value"].get("kind") == "PersistentVolume"
+                    and patch["value"].get("spec", {}).get("csi", {}).get("driver")
+                    == "blob.csi.azure.com"
+                    and endpoint_annotation
+                    in patch["value"].get("metadata", {}).get("annotations", {})
+                    and patch["value"]
+                    .get("spec", {})
+                    .get("csi", {})
+                    .get("volumeAttributes", {})
+                    .get("containerName")
```

Comment:
Would be nice if this if could be broken down to something more readable, like
```
is_persistent_volume = patch["value"].get("kind") == "PersistentVolume"
driver_is_azure = patch["value"].get("spec", {}).get("csi", {}).get("driver")== "blob.csi.azure.com"
has_container_name = patch["value"]
                    .get("spec", {})
                    .get("csi", {})
                    .get("volumeAttributes", {})
                    .get("containerName")

if isinstance(patch["value"], dict) and is_persistent_volume and driver_is_azure and has_container_name [...]:
```
then the if is a bit more readable. But not strictly necessary